src/transformers/pipelines/document_question_answering.py

```diff
@@ -0,0 +1,429 @@
+
+    def preprocess(
+        self,
+        input,
+        padding="do_not_pad",
+        doc_stride=None,
+        max_question_len=64,
+        max_seq_len=None,
+        word_boxes: Tuple[str, List[float]] = None,
+        lang=None,
+        tesseract_config="",
+    ):
+        # NOTE: This code mirrors the code in question answering and will be implemented in a follow up PR
+        # to support documents with enough tokens that overflow the model's window
+        #        if max_seq_len is None:
+        #            # TODO: LayoutLM's stride is 512 by default. Is it ok to use that as the min
+        #            # instead of 384 (which the QA model uses)?
+        #            max_seq_len = min(self.tokenizer.model_max_length, 512)
+
+        if doc_stride is not None:
+            # TODO implement
+            # doc_stride = min(max_seq_len // 2, 128)
+            raise ValueError("Unsupported: striding inputs")
+
+        image = None
+        image_features = {}
+        if "image" in input:
+            image = load_image(input["image"])
+            if self.feature_extractor is not None:
+                image_features.update(self.feature_extractor(images=image, return_tensors=self.framework))
+            elif isinstance(self.model, VisionEncoderDecoderModel):
+                raise ValueError("If you are using a VisionEncoderDecoderModel, you must provide a feature extractor")
+
+        words, boxes = None, None
+        if not isinstance(self.model, VisionEncoderDecoderModel):
+            if "word_boxes" in input:
+                words = [x[0] for x in input["word_boxes"]]
+                boxes = [x[1] for x in input["word_boxes"]]
+            elif "words" in image_features and "boxes" in image_features:
+                words = image_features.pop("words")
+                boxes = image_features.pop("boxes")
+            elif image is not None:
+                if not TESSERACT_LOADED:
+                    raise ValueError(
+                        "If you provide an image without word_boxes, then the pipeline will run OCR using Tesseract,"
+                        " but pytesseract is not available"
+                    )
+                if TESSERACT_LOADED:
+                    words, boxes = apply_tesseract(image, lang=lang, tesseract_config=tesseract_config)
+            else:
+                raise ValueError(
+                    "You must provide an image or word_boxes. If you provide an image, the pipeline will automatically"
+                    " run OCR to derive words and boxes"
+                )
+
+        if self.tokenizer.padding_side != "right":
+            raise ValueError(
+                "Document question answering only supports tokenizers whose padding side is 'right', not"
+                f" {self.tokenizer.padding_side}"
+            )
+
+        if isinstance(self.model, VisionEncoderDecoderModel):
+            task_prompt = f'<s_docvqa><s_question>{input["question"]}</s_question><s_answer>'
```

Comment:
That seems way too specific.

This seems like a model specific prompt that should be defined in the `tokenizer` of this particular model, no ?

Reply:
I think that's the right place, but as it stands, I do not believe it is encoded there. I based the logic from [here](https://huggingface.co/spaces/nielsr/donut-docvqa/blob/main/app.py). CC @NielsRogge

Reply:
No model has prompts defined within its tokenizer, so Donut is no exception. This is the prompt that the authors defined.

Reply:
Yes this is the Donut-specific prompt that the authors defined. No model has prompts like this defined within its tokenizer, unless it's `decoder_start_token_id`, which is defined in the configuration.

Reply:
In the text generation pipeline, I see that the `preprocess` method accepts a `prompt_text` argument.

Reply:
Can we refine the test to be more specific to Donut then? Vision encoder decoder is a wide class.

Reply:
Yeah, I'll add a check above where we determine if it's a `VisionEncoderDecoder` and raise a `ValueError` if it's not donut. I think when we add the next VisionEncoderDecoder, we can then tease apart the right abstraction (for stuff that applies to donut vs. any vision encoder/decoder).